source/sorted_table_model.ts

```diff
@@ -0,0 +1,198 @@
+import * as Kola from 'kola-signals';
+import { Comparator } from './comparator';
+import { AddRowOperation, MoveRowOperation, Operation, RemoveRowOperation,
+  Transaction, UpdateOperation } from './operations';
+import { SortOrder } from './sort_order';
+import {TableModel} from './table_model';
+import {TransactionLog} from './transaction_log';
+import {TranslatedTableModel} from './translated_table_model';
+
+/** A table that sorts columns. */
+export class SortedTableModel extends TableModel {
+
+  /**
+   * Constructs a model adapting an existing TableModel.
+   * @param model The TableModel to adapt.
+   */
+  constructor(model: TableModel) {
+    super();
+    this.comparator = new Comparator();
+    this.sortOrder = [];
+    for(let i = 0; i < model.columnCount; ++i) {
+      this.sortOrder.push(SortOrder.NONE);
+    }
+    this.sortPriority = [];
+    this.translatedTable = new TranslatedTableModel(model);
+    this.translatedTable.connect(this.handleSourceOperation);
+    this.movesToIgnore = 0;
+    this.transactionLog = new TransactionLog();
+  }
+
+  /** 
+   * Sorts the table using @param column as the column with the highest sort
+   * priority. All previous sortOrders remain and are pushed down a rank.
+   * @param column - The column that will now have the highest priority
+   * @param sortOrder - The sort order of column.
+   */
+  public updateSortOrder(column: number, sortOrder: SortOrder) {
+    if(column < 0 || this.columnCount <= column) {
+      throw new Error('The column is outside of range.');
+    } else if(sortOrder === SortOrder.UNSORTABLE) {
+      throw new Error('The column is unsortable.');
+    } else if(sortOrder === SortOrder.NONE) {
```

Comment:
Have setting the sort order to ```NONE``` or ```UNSORTABLE``` remove the column from the sort order instead of resulting in an error.